### ISSUE TITLE
Enemy movement needs work

### ISSUE THREAD

**Issue by ArifRoktim** [ArifRoktim/PotatoPower#1]
Enemies currently choose a target Tile and move toward that Tile.
Bug is: Enemies don't move around corners. When they reach the tile before a corner, they don't properly choose the next tile to target.

Diagnostic print statements are in Enemy.move(). Some of them are commented out.

**Comment by ArifRoktim** [comment]
Here is my planning for enemy movement:

```
Enemy moving:

If have target:
  _xPos += dx; _yPos += dy // move accordingly
  // if i reach my target, set target to null
  if( (int) _xPos == target._xPos &&
      (int) _yPos == target._yPos ){
    target = null; // <no longer have target>
  }

If have no Target:
  Make a new list
  Go through the four adjacent tiles
  If a tile has greater value than current tile, add that Tile to list
  Set target to random Tile from list
  < Now I have a target >
  Set x and y speeds to be in direction of target:
  dx = _d * (int) Math.signum( _xPos - target._xPos )
  dy = _d * (int) Math.signum( _yPos - target._yPos )
```

**Comment by ArifRoktim** [comment]
Enemy movement fixed by b06afb6df59558015f5529093f02f604af047885 but is very jittery
